src/main/java/com/ruralsourcing/WeatherBot/service/ApiTimerTask.java

```diff
@@ -0,0 +1,41 @@
+package com.ruralsourcing.WeatherBot.service;
+
+import com.google.gson.Gson;
+import com.ruralsourcing.WeatherBot.model.ResponseModel;
+
+import java.io.IOException;
+import java.util.TimerTask;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class ApiTimerTask extends TimerTask {
+
+    private final AtomicReference<ResponseModel> reference;
+
+    public ApiTimerTask(final AtomicReference<ResponseModel> reference) {
+
+        this.reference = reference;
+    }
+
+    @Override
+    public void run() {
+        reference.set(ApiCall());
+    }
+
+    public static ResponseModel ApiCall(){
```

Comment:
I was digging into this while figuring out how to use the config.properties file to hide the Mesonet API token, and I had it as a static method because it's what I use for the first initialization of the atomic reference that holds the API data, which I then pass into the API call thread and the message listener thread. By getting rid of apiCall() I'll have to initialize the reference using the default constructor. 

It could potentially cause an instance where if someone messages the slackbot immediately after it starts but before the threads are started then it might cause an issue where the reference isn't initialized. 

My first thought to solve that issue is to wait briefly after starting the API thread but before starting the message listener thread, but also since this program is intended to be running continually, it might not really be an issue in the first place. Hm.

Reply:
@DemonWav  I think this got fixed as part of commit 51e585bb but let me know what you think

Reply:
I've done another review which hits on this question.